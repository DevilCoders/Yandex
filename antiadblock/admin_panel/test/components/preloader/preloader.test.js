import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import Preloader from 'app/components/preloader/preloader';

describe('preloader component', () => {
    beforeAll(() => {
        console.error = error => { // eslint-disable-line no-console
            throw new Error(error);
        };
    });

    test('without props', () => {
        const wrapper = shallow(<Preloader />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('with fit prop', () => {
        const wrapper = shallow(<Preloader fit='viewport' />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('with invalid fit prop', () => {
        const wrapperFunction = () => shallow(<Preloader fit='window' />);
        expect(wrapperFunction).toThrowErrorMatchingSnapshot();
    });
});
