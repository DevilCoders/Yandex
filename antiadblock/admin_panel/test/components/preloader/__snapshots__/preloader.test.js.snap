// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preloader component with fit prop 1`] = `
<Bem
  block="preloader"
  mods={
    Object {
      "fit": "viewport",
    }
  }
>
  <spin2
    mix={
      Object {
        "block": "preloader",
        "elem": "spin",
      }
    }
    progress={true}
    size="m"
    tone="default"
    view="default"
  />
</Bem>
`;

exports[`preloader component with invalid fit prop 1`] = `
"Warning: Failed prop type: Invalid prop \`fit\` of value \`window\` supplied to \`Preloader\`, expected one of [\\"viewport\\"].
    in Preloader"
`;

exports[`preloader component without props 1`] = `
<Bem
  block="preloader"
  mods={
    Object {
      "fit": undefined,
    }
  }
>
  <spin2
    mix={
      Object {
        "block": "preloader",
        "elem": "spin",
      }
    }
    progress={true}
    size="m"
    tone="default"
    view="default"
  />
</Bem>
`;
